Fix Z-Wave JS cover stop support

Yukon opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This change fixes issues stopping covers (blinds) I was having with my home setup. The Z-Wave JS integration would blindly set the Up/Down property twice, causing the blinds to move again. The integration will now only stop one direction which will[ stop the cover](https://github.com/zwave-js/node-zwave-js/blob/1c2d267e0b4ad759681f3984de58c340b40b8a91/packages/cc/src/cc/MultilevelSwitchCC.ts#L444) regardless of actual direction of travel. 
## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [x] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

homeassistant commented:
Hi @Yukon, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#78723).

Once you do that we will be able to review and accept this pull request.

Thanks!

MartinHjelmare commented:
Please make sure the CI checks pass.

kpine commented on homeassistant/components/zwave_js/cover.py:
```diff
@@ -165,7 +165,13 @@ async def async_stop_cover(self, **kwargs: Any) -> None:
         )
-        if open_value:
+
+        if not open_value.metadata.readable:
```
What is the purpose of adding this check? All of these switch type properties, Up, Down, Close, Open, [are write-only](https://zwave-js.github.io/node-zwave-js/#/api/CCs/MultilevelSwitch?id=levelchangedownswitchtype-switchtype).

Yukon replied:
Thanks for the Z-Wave reference, in reviewing a prior fixture the state is mocked up as readable. https://github.com/home-assistant/core/blob/3a8282d0c5f7c2f72b677e84395543797418ba1a/tests/components/zwave_js/fixtures/cover_aeotec_nano_shutter_state.json#L214-L218

This may not actually be the case in the real work if your reference is the sole truth and the mock is incorrect.
@p4p3r  Do you have any experience with the Nano Shutter `Open`/`Close` state actually being readable during https://github.com/home-assistant/core/pull/52881 testing?

kpine replied:
The mock files do get out of date as the driver changes. The up/down values were made write-only in [this change](https://github.com/zwave-js/node-zwave-js/commit/603d27cf4450865ae1c3026b8fa6742ce551d62c).

Regardless, the read/write state of the value ID doesn't have anything to do with the functionality of the device. That is metadata for the value ID.

kpine reviewed (commented):
I don't think this is the right solution. I am fairly certain the root of this problem is it looks like the [original implementation](https://github.com/home-assistant/core/pull/45805) was copied from the [OZW integration code](https://github.com/home-assistant/core/blob/c95a72d2adc658a9b11ccbffbac1a4cea1b8b71d/homeassistant/components/ozw/cover.py#L83-L90), which sent multiple open/close commands to workaround a bug in OZW (legacy zwave used a different API so it doesn't have this same code).

The desired Z-Wave command to stop an ongoing multilevel-switch transition is the Multilevel Switch CC Stop Level Command. _Any one_ of the open/up/close/down values set to a `False` value will result in the [stop level command](https://github.com/zwave-js/node-zwave-js/blob/1c2d267e0b4ad759681f3984de58c340b40b8a91/packages/cc/src/cc/MultilevelSwitchCC.ts#L444). If there's an Up value, then there's always a Down, Open and Close, etc, so you just need to find one them, e.g. On, Up, Open, etc.

All possible switch types are listed [here](https://github.com/zwave-js/node-zwave-js/blob/0dfceeaaeeec3a7ea2a9dc5164e48257edecb91b/packages/cc/src/lib/_Types.ts#L1044-L1053). Each `X/Y` entry generates separate "X" and "Y" value IDs.

Another approach to finding a value would be to search all values and find the first that has metadata `cc_specific.switch_type` (or however the Python lib represents it), and use that one. Or, one could just use the CC API instead (see invoke_cc_api used elsewhere) and manually call the `stopLevelChange` instead, but you'd also need to check for support.

In summary, the solution should probably be to just remove the duplicate set_value. Your change effectively does this because it returns on the first check.

If you're unsure of how these properties are supposed to work (and don't take my word for it), I would confirm with the node-zwave-js project.

Yukon commented:
Hopefully I can get some feedback from others that have worked with covers about the `readable` state. It seems to be the case from the additional details @kpine provided that it really does not matter as setting the `Open` or `Close` state performs the same action in Wave-JS. I will jump on making that change, hopefully I will be able to figure out what is tripping up the flake8 CI check.

Yukon commented:
Changes are up! Much simpler approach calling `stopLevelChange`.

MartinHjelmare commented:
@kpine are you happy with this change?

MartinHjelmare commented:
@Yukon I'd suggest we follow the suggestion by @kpine and just remove the `close_value` call and adjust the tests accordingly. This will make the change smaller and should mostly be a deletion of code compared to the dev branch.

Yukon committed:
Fix zwave js cover stop support

Only set one cover property that will send the Multilevel Switch CC Stop Level command.

Yukon commented:
@MartinHjelmare The changes have been simplified to just remove the problematic extra stop.

MartinHjelmare commented:
Thanks! Please solve the lint errors.

frenck committed:
Fix linter warnings

frenck commented:
Pushed a fix for the linter warnings

MartinHjelmare reviewed (approved):
Thanks!
